src/AssemblyAI/Files/FilesClient.cs

```diff
@@ -19,21 +20,59 @@ public FilesClient(RawClient client)
     /// <summary>
     /// Upload a media file to AssemblyAI's servers.
     /// </summary>
-    public async Task<UploadedFile> UploadAsync(Stream request)
+    public async Task<UploadedFile> UploadAsync(Stream request, RequestOptions? options = null)
     {
         var response = await _client.MakeRequestAsync(
             new RawClient.StreamApiRequest
             {
+                BaseUrl = _client.Options.BaseUrl,
                 Method = HttpMethod.Post,
                 Path = "v2/upload",
-                Body = request
+                Body = request,
+                Options = options
             }
         );
         var responseBody = await response.Raw.Content.ReadAsStringAsync();
         if (response.StatusCode is >= 200 and < 400)
         {
-            return JsonSerializer.Deserialize<UploadedFile>(responseBody)!;
+            try
+            {
+                return JsonUtils.Deserialize<UploadedFile>(responseBody)!;
+            }
+            catch (JsonException e)
+            {
+                throw new AssemblyAIClientException("Failed to deserialize response", e);
+            }
+        }
+
+        try
+        {
+            switch (response.StatusCode)
+            {
+                case 400:
+                    throw new BadRequestError(JsonUtils.Deserialize<Error>(responseBody));
```

Comment:
These exception classes should be suffixed by `Exception`.
However, we could just have a single exception for all of these.
We don't need a separate exception class for each error.